components/search/SearchCollection.vue

```diff
@@ -45,162 +45,142 @@
+      break
+    case 'e':
+      updateSortBy(props.sortOption?.[3] || '')
+      break
   }
+}
 
-  get searchQuery(): string {
-    return this.search
-  }
+const vListed = computed({
+  get: () => props.listed,
+  set: (value: boolean) => {
+    updateListed(value)
+  },
+})
 
-  set searchQuery(value: string) {
-    this.updateSearch(value)
-  }
+const vOwned = computed({
+  get: () => props.owned,
+  set: (value: boolean) => {
+    updateOwned(value)
+  },
+})
 
-  get typeQuery(): string {
-    return this.type
-  }
+const searchQuery = computed({
+  get: (): string => props.search,
+  set: (value: string) => {
+    updateSearch(value)
+  },
+})
 
-  set typeQuery(value: string) {
-    this.updateType(value)
-  }
+const replaceBuyNowWithYolo = preferencesStore.getReplaceBuyNowWithYolo
 
-  get replaceBuyNowWithYolo(): boolean {
-    return this.preferencesStore.getReplaceBuyNowWithYolo
-  }
+const updateListed = useDebounceFn((value: string | boolean) => {
+  const queryValue = String(value) === 'true'
+  replaceUrl(queryValue, 'listed')
+  emit('update:listed', queryValue)
+}, 50)
 
-  @Emit('update:listed')
-  @Debounce(50)
-  updateListed(value: string | boolean): boolean {
-    const queryValue = String(value) === 'true'
-    this.replaceUrl(queryValue, 'listed')
-    return queryValue
-  }
+const updateOwned = useDebounceFn((value: string | boolean) => {
+  const queryValue = value ? String(value) : ''
+  replaceUrl(queryValue, 'owned')
+  emit('update:owned', Boolean(queryValue))
+}, 50)
 
-  @Emit('update:owned')
-  @Debounce(50)
-  updateOwned(value: string | boolean): boolean {
-    const queryValue = value ? String(value) : ''
-    this.replaceUrl(queryValue, 'owned')
-    return Boolean(queryValue)
-  }
+const updateType = useDebounceFn((value: string) => {
+  replaceUrl(value, 'type')
+  emit('update:type', value)
+}, 50)
 
-  @Emit('update:type')
-  @Debounce(50)
-  updateType(value: string): string {
-    this.replaceUrl(value, 'type')
-    return value
+const updateSortBy = useDebounceFn((value: string) => {
+  const listed = Boolean(value?.toLowerCase().indexOf('price') > -1)
+  if (listed && !vListed.value) {
+    vListed.value = true
   }
 
-  @Emit('update:sortBy')
-  @Debounce(400)
-  updateSortBy(value: string): string {
-    const listed = Boolean(value?.toLowerCase().indexOf('price') > -1)
-    if (listed && !this.vListed) {
-      this.vListed = true
-    }
+  replaceUrl(value, 'sort')
+  emit('update:sortBy', value)
+}, 400)
 
-    this.replaceUrl(value, 'sort')
-    return value
+const updateSearch = useDebounceFn((value: string) => {
+  if (value !== searchQuery.value) {
+    replaceUrl(value)
+    emit('update:search', value)
   }
+}, 400)
 
-  @Emit('update:search')
-  @Debounce(400)
-  updateSearch(value: string): string {
-    value !== this.searchQuery && this.replaceUrl(value)
-    return value
+const replaceUrl = useDebounceFn((value: boolean | string, key = 'search') => {
```

Comment:
we have `useReplaceUrl()` composable for that

Reply:
Noted on this. Will refactor it using the composable.